server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexServiceTests.java

```diff
@@ -267,7 +268,7 @@ public void testPrepareResizeIndexSettings() {
                             settings.get("index.analysis.analyzer.default.tokenizer"),
                             equalTo("keyword"));
                     assertThat(settings.get("index.routing.allocation.initial_recovery._id"), equalTo("node1"));
-                    assertThat(settings.get("index.allocation.max_retries"), equalTo("1"));
+                    assertThat(settings.get("index.allocation.max_retries"), nullValue());
```

Comment:
Let's say we keep the production implementation as-is.

Do we need to continue to assert on this at all? It feels odd to single this setting as unset amongst all the other settings, it's only an accident of history that we're making an assertion here. Also, we would inherit this setting if it was present on an existing index (and an open question whether or not we should do that, I think we should, but I'm curious to hear your perspective) so this assertion can be misleading then (even though it's true in this test, because the index wasn't set up with this setting set).

If we do change the production implementation, to override any existing value for `index.allocation.max_retries`, then the assertion makes sense, but we should also consider covering the case where the setting is present on the source index.

Reply:
I edited the above comment to add a key missing not. To be clear, I think that we should *not* inherit any existing value for this setting, but instead override it. We have a built-in mechanism for this, to set not copyable on resize.

Reply:
And I don't think we need to worry about whether or not this a breaking change. Today we are universally overriding the value of the setting to one (whether it's set or unset). With the change as implemented, we would now inherit the value. If we change the production implementation to never copy the value for this setting, that wouldn't be breaking, since we weren't preserving it already today.

Reply:
I did think about removing that assertion, but also felt like I then did not really test the change. Ideally the test would test that only expected properties are present instead and I could make that change instead, as well as ensure it now copies it from source if set (depending on the copy-setting discussion).

About copying the setting, I think we should. I believe that is the intended direction after #30255? If this setting had been set explicitly on the source, I think it makes sense to carry it through shrinking, at least in an ILM context. Additionally, this would AFAICS be the only setting that would not be copied if we add this. I am curious to know why you are considering to not copy this specific setting?

Reply:
I think it's a setting of dubious value, and I'm not sure why we even have it (at the index level). We are already overriding it in resizes today, and we haven't seen complaints about that. We don't replicate it in cross-cluster replication.